saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -556,6 +572,8 @@ public Saml2MessageBinding getSingleSignOnServiceBinding() {
 
 			private Saml2MessageBinding singleSignOnServiceBinding = Saml2MessageBinding.REDIRECT;
 
+			private List<String> signAlgorithmUris = Arrays.asList(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256);
```

Comment:
I think it would probably be best to leave this empty from the start and then add it in the `build` method if it's still empty.

The reason for that is that this will likely be called by `OpenSamlAssertingPartyMetadataConverter`, which will take the list of signing algorithms from the asserting party's metadata and do something like:

```java
builder.assertingPartyDetails((details) -> details
    .signingMethodAlgorithms((a) -> a.addAll(algorithms))
);
```

If there's something already in the list, then it would need to also first clear the original list:

```java
builder.assertingPartyDetails((details) -> details
    .signingMethodAlgorithms((a) -> {
        a.clear();
        a.addAll(algorithms);
    })
);
```